src/components/App.jsx

```diff
@@ -168,16 +156,15 @@ function App() {
         setCurrentUser(user);
         sessionStorage.setItem('currentUser', JSON.stringify(currentUser));
 
-        const filterPredictions = Array.isArray(predictions) ? predictions.filter(prediction => {
+        const filterPredictions = Array.isArray(allPredictions) ? allPredictions.filter(prediction => {
```

Comment:
usually I use the original variable (e.g. `allPredictions`) instead of reassigning it to another variable `predictions` to reduce complexity & avoid careless mistakes due to variable namings